lumicks/pylake/tests/test_channels.py

```diff
@@ -106,6 +155,168 @@ def test_calibration_continuous_channels():
+    assert np.allclose((scc1 / [2, 2, 2, 2, 2]).data, [.5, 1, 1.5, 2, 2.5])
+
+    # Tests with raw numpy array
+    assert np.allclose((scc1 * np.array([2, 2, 2, 2, 2])).data, [2, 4, 6, 8, 10])
+    assert np.allclose((scc1 + np.array([2, 2, 2, 2, 2])).data, [3, 4, 5, 6, 7])
+    assert np.allclose((scc1 - np.array([2, 2, 2, 2, 2])).data, [-1, 0, 1, 2, 3])
+    assert np.allclose((scc1 / np.array([2, 2, 2, 2, 2])).data, [.5, 1, 1.5, 2, 2.5])
+
+    scc1 = channel.Slice(channel.TimeSeries([1, 2, 3, 4, 5], [1, 2, 3, 4, 5]))
+    scc2 = channel.Slice(channel.TimeSeries([1, 1, 3, 1, 5], [1, 2, 3, 4, 5]))
+    scc3 = channel.Slice(channel.TimeSeries([1, 1, 3, 1, 5, 6], [1, 2, 3, 4, 5, 6]))
+
+    assert np.allclose((scc1 - scc2).data, [0, 1, 0, 3, 0])
+    assert np.allclose((scc1 + scc2).data, [2, 3, 6, 5, 10])
+    assert np.allclose((scc1 * scc2).data, [1, 2, 9, 4, 25])
+    assert np.allclose((scc1 / scc2).data, [1, 2, 1, 4, 1])
+    with pytest.raises(ValueError):
+        scc1 - scc3
+
+    # Tests using arithmetic with a scalar
+    assert np.allclose((scc1 * 2).data, [2, 4, 6, 8, 10])
+    assert np.allclose((scc1 + 2).data, [3, 4, 5, 6, 7])
+    assert np.allclose((scc1 - 2).data, [-1, 0, 1, 2, 3])
+    assert np.allclose((scc1 / 2).data, [.5, 1, 1.5, 2, 2.5])
+    assert np.allclose((2 * scc1).data, [2, 4, 6, 8, 10])
+    assert np.allclose((2 + scc1).data, [3, 4, 5, 6, 7])
+
+    # Tests with raw list
+    assert np.allclose((scc1 * [2, 2, 2, 2, 2]).data, [2, 4, 6, 8, 10])
+    assert np.allclose((scc1 + [2, 2, 2, 2, 2]).data, [3, 4, 5, 6, 7])
+    assert np.allclose((scc1 - [2, 2, 2, 2, 2]).data, [-1, 0, 1, 2, 3])
+    assert np.allclose((scc1 / [2, 2, 2, 2, 2]).data, [.5, 1, 1.5, 2, 2.5])
+
+    # Tests with raw list
+    assert np.allclose(([2, 2, 2, 2, 2] * scc1).data, [2*1, 2*2, 2*3, 2*4, 2*5])
+    assert np.allclose(([2, 2, 2, 2, 2] + scc1).data, [2+1, 2+2, 2+3, 2+4, 2+5])
+    assert np.allclose(([2, 2, 2, 2, 2] / scc1).data, [2/1, 2/2, 2/3, 2/4, 2/5])
+    assert np.allclose(([2, 2, 2, 2, 2] - scc1).data, [2-1, 2-2, 2-3, 2-4, 2-5])
+
+    # Tests with raw numpy array
+    assert np.allclose((scc1 * np.array([2, 2, 2, 2, 2])).data, [2, 4, 6, 8, 10])
+    assert np.allclose((scc1 + np.array([2, 2, 2, 2, 2])).data, [3, 4, 5, 6, 7])
+    assert np.allclose((scc1 - np.array([2, 2, 2, 2, 2])).data, [-1, 0, 1, 2, 3])
+    assert np.allclose((scc1 / np.array([2, 2, 2, 2, 2])).data, [.5, 1, 1.5, 2, 2.5])
+
+    # In principle cross-class operators should be allowed as long as the units are compatible
+    scc1 = channel.Slice(channel.Continuous([1, 2, 3, 4, 5], start=0, dt=1))
+    assert np.allclose((scc1 - scc2).data, [0, 1, 0, 3, 0])
+    assert np.allclose((scc1 + scc2).data, [2, 3, 6, 5, 10])
+    assert np.allclose((scc1 * scc2).data, [1, 2, 9, 4, 25])
+    assert np.allclose((scc1 / scc2).data, [1, 2, 1, 4, 1])
+    with pytest.raises(ValueError):
+        scc1 - scc3
+
+
+def test_arithmetic_operations():
```

Comment:
A function so nice you wrote it twice?

I'm confused. :thinking:

Reply:
Huh... 😕 You and me both. My best guess is that I must have done this accidentally when I rebased it. Removed the duplication now.